discovery/chan_series.go

```diff
@@ -178,8 +191,13 @@ func (c *ChanSeries) UpdatesInHorizon(chain chainhash.Hash,
 	if err != nil {
 		return nil, err
 	}
+
 	for _, nodeAnn := range nodeAnnsInHorizon {
-		nodeAnn := nodeAnn
+		// If this node has not been seen in the above channels, we can
+		// skip sending its NodeAnnouncement.
+		if _, seen := nodesFromChan[nodeAnn.PubKeyBytes]; !seen {
```

Comment:
Have you seen us send these node ann's in the wild? Each block we should be pruning node anns that don't have any active channels.